cmd/traffic_ctl/traffic_ctl.cc

```diff
@@ -222,7 +222,7 @@ main(int argc, const char **argv)
 
   ArgumentDescription argument_descriptions[] = {
     {"debug", '-', "Enable debugging output", "F", &debug, nullptr, nullptr},
-    HELP_ARGUMENT_DESCRIPTION(),
+    // HELP_ARGUMENT_DESCRIPTION(), //quick fix to get subcommands to show up when -h or --help arguments given
```

Comment:
If this is the right fix, we ought to just nuke the line honestly.

Reply:
Probably better to have a custom implementation of the `--help` flag so that it shows up in the help output itself:
```
diff --git a/cmd/traffic_ctl/traffic_ctl.cc b/cmd/traffic_ctl/traffic_ctl.cc
index 32f59d94a..46062f792 100644
--- a/cmd/traffic_ctl/traffic_ctl.cc
+++ b/cmd/traffic_ctl/traffic_ctl.cc
@@ -211,6 +211,15 @@ CtrlGenericSubcommand(const char *name, const subcommand *cmds, unsigned ncmds,
   return CtrlSubcommandUsage(name, cmds, ncmds, nullptr, 0);
 }
 
+static const subcommand commands[] = {
+  {subcommand_alarm, "alarm", "Manipulate alarms"},
+  {subcommand_config, "config", "Manipulate configuration records"},
+  {subcommand_metric, "metric", "Manipulate performance metrics"},
+  {subcommand_server, "server", "Stop, restart and examine the server"},
+  {subcommand_storage, "storage", "Manipulate cache storage"},
+  {subcommand_plugin, "plugin", "Interact with plugins"},
+};
+
 int
 main(int argc, const char **argv)
 {
@@ -220,22 +229,17 @@ main(int argc, const char **argv)
   CtrlVersionInfo.setup(PACKAGE_NAME, "traffic_ctl", PACKAGE_VERSION, __DATE__, __TIME__, BUILD_MACHINE, BUILD_PERSON, "");
   program_name = CtrlVersionInfo.AppStr;
 
-  ArgumentDescription argument_descriptions[] = {
+  const ArgumentDescription argument_descriptions[] = {
     {"debug", '-', "Enable debugging output", "F", &debug, nullptr, nullptr},
-    HELP_ARGUMENT_DESCRIPTION(),
+    {"help", 'h', "Print usage information", nullptr, nullptr, nullptr,
+      [](const ArgumentDescription *args, unsigned nargs, const char *arg_unused) {
+        CtrlSubcommandUsage(nullptr, commands, countof(commands), args, nargs);
+      }
+    },
     VERSION_ARGUMENT_DESCRIPTION(),
     RUNROOT_ARGUMENT_DESCRIPTION(),
   };
 
-  const subcommand commands[] = {
-    {subcommand_alarm, "alarm", "Manipulate alarms"},
-    {subcommand_config, "config", "Manipulate configuration records"},
-    {subcommand_metric, "metric", "Manipulate performance metrics"},
-    {subcommand_server, "server", "Stop, restart and examine the server"},
-    {subcommand_storage, "storage", "Manipulate cache storage"},
-    {subcommand_plugin, "plugin", "Interact with plugins"},
-  };
-
   BaseLogFile *base_log_file = new BaseLogFile("stderr");
   diags                      = new Diags(program_name, "" /* tags */, "" /* actions */, base_log_file);
 ```